samples: cellular: nrf_cloud_multi_service: Add minimal CoAP and MQTT overlays (#12492)
nrfconnect/sdk-nrf: samples/cellular/nrf_cloud_multi_service/README.rst

The following key features of this sample may be independently disabled:

* GNSS-based location tracking - by setting the :kconfig:option:`CONFIG_LOCATION_TRACKING_GNSS` option to disabled.
* Cellular-based location tracking - by setting the :kconfig:option:`CONFIG_LOCATION_TRACKING_CELLULAR` option to disabled.
* Wi-Fi-based location tracking - by setting the :kconfig:option:`CONFIG_LOCATION_TRACKING_WIFI` option to disabled.
* Temperature tracking - by setting the :kconfig:option:`CONFIG_TEMP_TRACKING` option to disabled.
* GNSS assistance (A-GPS) - by setting the :kconfig:option:`CONFIG_NRF_CLOUD_AGPS` option to disabled.
* Predictive GNSS assistance (P-GPS) - by setting the :kconfig:option:`CONFIG_NRF_CLOUD_PGPS` option to disabled.
* FOTA when using MQTT - by setting the :kconfig:option:`CONFIG_NRF_CLOUD_FOTA` option to disabled.

> * FOTA when using CoAP - by setting the :kconfig:option:`COAP_FOTA` option to disabled.
> * Shadow handling when using CoAP - by setting the :kconfig:option:`COAP_SHADOW` option to disabled.

Reviewer: ```suggestion
* FOTA when using CoAP - by setting the :ref:`CONFIG_COAP_FOTA <CONFIG_COAP_FOTA>` option to disabled.
* Shadow handling when using CoAP - by setting the :ref:`CONFIG_COAP_SHADOW <CONFIG_COAP_SHADOW>` option to disabled.
```

Author: Why is this better?  I actually changed lines 352-355 to use :kconfig:option: instead, because I did not see any difference in appearance in the preview form of the docs from the longer :ref:`name <name>` form.

Reviewer 2: I am also a bit confused, I'm the one who mentioned Pete should change it, because I thought we were spposed to be using :kconfig:option: for kconfigs, and all the options further down the list use it

If you disable GNSS, Wi-Fi-based, and cellular-based location tracking, location tracking is completely disabled.

> In that case, also set the :ref:`CONFIG_LOCATION_TRACKING` option to disabled.

Reviewer: ```suggestion
In that case, also set the :ref:`CONFIG_LOCATION_TRACKING <CONFIG_LOCATION_TRACKING>` option to disabled.
```

> For examples, see the related minimal overlays below: :ref:`nrf_cloud_multi_service_minimal`.

Reviewer: ```suggestion
For examples, see the related minimal overlays in the :ref:`nrf_cloud_multi_service_minimal` section.
```

   Sets the GNSS fix timeout in seconds.
   On each location sample, try for this long to achieve a GNSS fix before falling back to cellular positioning.

.. _CONFIG_LOCATION_TRACKING:

CONFIG_LOCATION_TRACKING - Enable or disable location tracking
   Enables location tracking.

>    Enable at least one location tracking method or a build error will occur.

Reviewer: ```suggestion
   Enable at least one location tracking method to avoid a build error.
```

CONFIG_LOCATION_TRACKING_GNSS - GNSS location tracking
   Enables GNSS location tracking.

>    Disable :ref:`CONFIG_LOCATION_TRACKING` and all location tracking methods to completely disable location tracking.

Reviewer: ```suggestion
   Disable :ref:`CONFIG_LOCATION_TRACKING <CONFIG_LOCATION_TRACKING>` and all location tracking methods to completely disable location tracking.
```

   Defaults to enabled.

.. _CONFIG_LOCATION_TRACKING_CELLULAR:

CONFIG_LOCATION_TRACKING_CELLULAR - Cellular location tracking
   Enables cellular location tracking.

>    Disable :ref:`CONFIG_LOCATION_TRACKING` and all location tracking methods to completely disable location tracking.

Reviewer: ```suggestion
   Disable :ref:`CONFIG_LOCATION_TRACKING <CONFIG_LOCATION_TRACKING>` and all location tracking methods to completely disable location tracking.
```

   Defaults to enabled.

.. _CONFIG_LOCATION_TRACKING_WIFI:

CONFIG_LOCATION_TRACKING_WIFI - Wi-Fi location tracking
   Enables Wi-Fi location tracking.

>    Disable :ref:`CONFIG_LOCATION_TRACKING` and all location tracking methods to completely disable location tracking.

Reviewer: ```suggestion
   Disable :ref:`CONFIG_LOCATION_TRACKING <CONFIG_LOCATION_TRACKING>` and all location tracking methods to completely disable location tracking.
```

When using CoAP, the following additional configuration options are available:

.. _CONFIG_COAP_SHADOW:

CONFIG_COAP_SHADOW - Enable shadow handling
   Periodically check for and process shadow delta messages.

> If :kconfig:option:`CONFIG_COAP_SHADOW` is enabled, these options additional are available:

Reviewer: ```suggestion
If :ref:`CONFIG_COAP_SHADOW <CONFIG_COAP_SHADOW>` is enabled, these options additional are available:
```

CONFIG_COAP_SHADOW_THREAD_STACK_SIZE - CoAP Shadow Thread Stack Size (bytes)
   Sets the stack size (in bytes) for the shadow delta checking thread of the sample.

.. _CONFIG_COAP_FOTA:

CONFIG_COAP_FOTA - Enable FOTA with CoAP
   The sample periodically checks for pending FOTA jobs.
   The sample performs the FOTA update when received.

> If :kconfig:option:`CONFIG_COAP_FOTA` is enabled, these options additional are available:

Reviewer: ```suggestion
If :ref:`CONFIG_COAP_FOTA <CONFIG_COAP_FOTA>` is enabled, these options additional are available:
```

Deselect the :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT` Kconfig option and select :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY` instead.
See `Dictionary-based Logging`_ to learn how dictionary-based logging works, how the dictionary is built, and how to decode the binary log output.

.. _nrf_cloud_multi_service_minimal:

Building with minimal services
==============================

To build the sample with only temperature tracking enabled for either MQTT or CoAP, add the following parameter to your build command:

> ``-DOVERLAY_CONFIG=overlay_min_mqtt.conf``
> or
> ``-DOVERLAY_CONFIG=overlay_min_coap.conf``

Reviewer: ```suggestion
``-DOVERLAY_CONFIG=overlay_min_mqtt.conf``

or

``-DOVERLAY_CONFIG=overlay_min_coap.conf``
```
